applications/ShapeOptimizationApplication/python_scripts/custom_math.py

```diff
@@ -0,0 +1,422 @@
+# ==============================================================================
+#  KratosShapeOptimizationApplication
+#
+#  License:         BSD License
+#                   license: ShapeOptimizationApplication/license.txt
+#
+#  Main authors:    Baumgaertner Daniel, https://github.com/dbaumgaertner
+#
+# ==============================================================================
+
+# Making KratosMultiphysics backward compatible with python 2.6 and 2.7
+from __future__ import print_function, absolute_import, division
+
+# Import additional libraries
+import math
+
+# ==============================================================================
+def SafeConvertVectorToMatrix(_L):
+    if IsVector(_L):
+        return [_L]
+    else:
+        return _L
+
+# ------------------------------------------------------------------------------
+def IsVector(_X):
+    if len(_X)==0 or isinstance(_X[0],(float,int)):
```

Comment:
strange check, is this really doing what the name suggests

Reply:
Yes, even if it is not a very nice way of doing it. But this is going to be solved, once some parts of the trust region algorithm go down to C++, where we can use the boos linear algebra things and things get more descriptive.